Flaky test KubernetesValidationHookHandlingTest.validatingWebhookRejectsInvalidResource

```
Error:  Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 4.836 s <<< FAILURE! -- in io.fabric8.kubeapitest.sample.KubernetesValidationHookHandlingTest
Error:  io.fabric8.kubeapitest.sample.KubernetesValidationHookHandlingTest.validatingWebhookRejectsInvalidResource -- Time elapsed: 0.166 s <<< FAILURE!
java.lang.AssertionError: 

Expecting code to raise a throwable.
	at io.fabric8.kubeapitest.sample.KubernetesValidationHookHandlingTest.validatingWebhookRejectsInvalidResource(KubernetesValidationHookHandlingTest.java:106)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)

[INFO] Running io.fabric8.kubeapitest.UtilsTest
[INFO] Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.010 s -- in io.fabric8.kubeapitest.UtilsTest
[INFO] 
[INFO] Results:
[INFO] 
Error:  Failures: 
Error:    KubernetesValidationHookHandlingTest.validatingWebhookRejectsInvalidResource:106 
Expecting code to raise a throwable.
[INFO] 
Error:  Tests run: 25, Failures: 1, Errors: 0, Skipped: 0
```
Complete log file at https://productionresultssa16.blob.core.windows.net/actions-results/602e7785-fc16-44d3-9421-de5c0fe7b726/workflow-job-run-534527a0-90e7-5559-8a9e-6d9f534dd1df/logs/job/job-logs.txt?rsct=text%2Fplain&se=2026-05-05T09%3A21%3A30Z&sig=mgWJFbVQK%2BFjIRDzbV1snQFgBj2lHeAocGJ1fRADPvA%3D&ske=2026-05-05T13%3A00%3A33Z&skoid=ca7593d4-ee42-46cd-af88-8b886a2f84eb&sks=b&skt=2026-05-05T09%3A00%3A33Z&sktid=398a6654-997b-47e9-b12b-9515b896b4de&skv=2025-11-05&sp=r&spr=https&sr=b&st=2026-05-05T09%3A11%3A25Z&sv=2025-11-05

--- a/junit/kube-api-test/core/src/test/java/io/fabric8/kubeapitest/sample/KubernetesValidationHookHandlingTest.java
+++ b/junit/kube-api-test/core/src/test/java/io/fabric8/kubeapitest/sample/KubernetesValidationHookHandlingTest.java
@@ -41,12 +41,15 @@ import io.fabric8.mockwebserver.MockWebServer;
 import io.fabric8.mockwebserver.http.Dispatcher;
 import io.fabric8.mockwebserver.http.MockResponse;
 import io.fabric8.mockwebserver.http.RecordedRequest;
+import org.awaitility.Awaitility;
 import org.junit.jupiter.api.AfterAll;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.time.Duration;
+import java.util.UUID;
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
@@ -102,10 +105,23 @@ class KubernetesValidationHookHandlingTest {
     KubernetesClient client = new KubernetesClientBuilder().withConfig(Config.fromKubeconfig(kubeConfig)).build();
     applyConfig(client);
 
-    NamespaceableResource<Ingress> ingressResource = client.resource(invalidIngress());
-    assertThatThrownBy(ingressResource::create)
-        .isInstanceOf(KubernetesClientException.class)
-        .hasMessageContaining("Ingress with annotation 'reject=true' is not allowed");
+    // Webhook activation has a brief propagation delay after applyConfig; poll with a fresh
+    // resource name each attempt so a slipped-through create on an early poll does not
+    // shadow later attempts via AlreadyExists.
+    Awaitility.await()
+        .atMost(Duration.ofSeconds(30))
+        .pollInterval(Duration.ofMillis(200))
+        .untilAsserted(() -> {
+          Ingress invalid = invalidIngress("invalid-ingress-" + UUID.randomUUID());
+          NamespaceableResource<Ingress> ingressResource = client.resource(invalid);
+          try {
+            assertThatThrownBy(ingressResource::create)
+                .isInstanceOf(KubernetesClientException.class)
+                .hasMessageContaining("Ingress with annotation 'reject=true' is not allowed");
+          } finally {
+            client.resource(invalid).delete();
+          }
+        });
   }
 
   static MockResponse handleValidatingWebhook(RecordedRequest request) {
@@ -183,10 +199,10 @@ class KubernetesValidationHookHandlingTest {
         .build();
   }
 
-  public static Ingress invalidIngress() {
+  public static Ingress invalidIngress(String name) {
     return new IngressBuilder()
         .withNewMetadata()
-        .withName("invalid-ingress")
+        .withName(name)
         .addToAnnotations("reject", "true")
         .endMetadata()
         .withSpec(new IngressSpecBuilder()

